src/v/cluster/cluster_discovery.cc

```diff
@@ -69,22 +71,34 @@ ss::future<node_id> cluster_discovery::determine_node_id() {
     co_return assigned_node_id;
 }
 
-std::vector<model::broker> cluster_discovery::initial_seed_brokers() const {
+ss::future<cluster_discovery::brokers>
+cluster_discovery::initial_seed_brokers() {
     // If configured as the root node, we'll want to start the cluster with
     // just this node as the initial seed.
     if (config::node().empty_seed_starts_cluster()) {
         if (config::node().seed_servers().empty()) {
-            return {make_self_broker(config::node())};
+            co_return brokers{make_self_broker(config::node())};
         }
         // Not a root
-        return {};
+        co_return brokers{};
     }
     if (get_node_index_in_seed_servers()) {
```

Comment:
nit: IMO this is awkward to reuse here, because we're not using the returned value `idx`. It makes me wonder if we can replace it with some `is_configured_seed_server()` method that we can maybe cache once. Or maybe we should keep the seed servers around as a set and use some `seed_servers.contains(self)` style of checking

Reply:
Well `idx` is not used, however the complexy to obtain `idx` is the same as to obtain the flag whether we are a part of `seed_servers` at all. Caching that value is possible yet I don't think we can gain anything visible from it.

Reply:
My motivation is to make the code easier to understand upon first glance. IMO it's very obvious what it means for this node to be configured as a seed server -- it's less obvious what it means to have `get_node_index..()` return `nullopt`.

You're right, there is not functional change though, so feel free to punt.